### PR TITLE
Add ability to reference local resources for airplane dev

### PR DESCRIPTION
**Pull request by calufornia**
## Description

This PR adds the notion of an airplane dev config file, which will be used to store local resource configuration. The dev config will store a mapping from slug to resource configuration - this is because the task definition file stores resource attachments in the form of a map between alias to slug, and so combining the two we have a mapping from alias to resource configuration, which is what is required by our SDK. We add logic to encode resource configs into an `AIRPLANE_RESOURCES` env var, which will then be used by the Node SDK to get a task's resource attachments and builtins.

This PR also copies over many of the builtins structs - these should eventually go into their own repo but are included here for now. 

## Test plan
Tested with the below dev config and verified that resources were serialized into the `AIRPLANE_RESOURCES` env var properly:
```
env:
  API_KEY: asdf
  IMAGE: 1234
resources:
  db:
    kind: postgres
    slug: db
    username: admin
    password: airplane
  rest:
    kind: rest
    slug: rest
    baseURL: "localhost:1234"
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.18
 require (
 	github.com/AlecAivazis/survey/v2 v2.3.5
 	github.com/MakeNowJust/heredoc v1.0.0
-	github.com/airplanedev/lib v0.5.25
+	github.com/airplanedev/lib v0.5.25-0.20220714001423-df759db6bb66
```
Comment (calufornia) [inline go.mod:8]: Will update this before landing

File: pkg/conf/dev.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	Resources        map[string]map[string]interface{} `yaml:"resources"`
+	DecodedResources map[string]resource.Resource
+}
+
+func (d *DevConfig) DefaultPath() string {
+	homedir, err := os.UserHomeDir()
+	if err != nil {
+		panic("$HOME environment variable must be set")
+	}
+	return filepath.Join(
```
Comment (calufornia) [inline pkg/conf/dev.go:25]: Default filepath is still TBD

File: pkg/resource/postgres.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package resource
```
Comment (calufornia) [inline pkg/resource/postgres.go:1]: I've only added postgres for now, but assuming everything else looks good, will add all the other resource structs

**Review by calufornia — COMMENTED** [review]
Tests to be added

File: pkg/resource/postgres.go (hunk 3 above)
Comment (kellymtrinh) [reply pkg/resource/postgres.go:1]: I pretty much need these too for constructing the built in requests :/ maybe we can start with just moving the resource factories and clients out into a built-ins repo

File: pkg/conf/dev.go (hunk 2 above)
Comment (calufornia) [reply pkg/conf/dev.go]: Decided offline that there won't be a default filepath - users will specify a path when doing `airplane dev`

**Review by kellymtrinh — APPROVED** [review]

**Pull request merged**